src/test/javascript/spec/component/assessment-shared/assesment-filter-component.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { ArtemisTestModule } from '../../test.module';
+import { AssessmentFiltersComponent } from 'app/assessment/assessment-filters/assessment-filters.component';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ArtemisTranslatePipe } from 'app/shared/pipes/artemis-translate.pipe';
+import { MockDirective, MockPipe } from 'ng-mocks';
+import { NgModel } from '@angular/forms';
+import { TextSubmission } from 'app/entities/text-submission.model';
+import { Submission, SubmissionExerciseType } from 'app/entities/submission.model';
+import { Result } from 'app/entities/result.model';
+import dayjs from 'dayjs/esm';
+
+describe('AssessmentFiltersComponent', () => {
+    let component: AssessmentFiltersComponent;
+    let fixture: ComponentFixture<AssessmentFiltersComponent>;
+    let textSubmissionWithLock: TextSubmission;
+    let textSubmissionWithoutResult: TextSubmission;
+    let textSubmissionWithResult: TextSubmission;
+    let submissions: Submission[];
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            declarations: [AssessmentFiltersComponent, MockPipe(ArtemisTranslatePipe), MockDirective(NgModel)],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(AssessmentFiltersComponent);
+                component = fixture.componentInstance;
+                fixture.detectChanges();
+                textSubmissionWithLock = { id: 23, participation: { exercise: { id: 1 }, id: 2 }, submissionExerciseType: SubmissionExerciseType.TEXT } as TextSubmission;
+                textSubmissionWithoutResult = { id: 24, participation: { exercise: { id: 1 }, id: 3 }, submissionExerciseType: SubmissionExerciseType.TEXT } as TextSubmission;
+                textSubmissionWithResult = { id: 25, participation: { exercise: { id: 1 }, id: 4 }, submissionExerciseType: SubmissionExerciseType.TEXT } as TextSubmission;
+                submissions = [textSubmissionWithLock, textSubmissionWithoutResult, textSubmissionWithResult];
```

Comment:
Just something to think about 🙂 : This *could* be misleading here, as in these lines, the submissions don't actually have a lock or a result; yet, the variables are named like they did.

What about just moving these definitions down to the second test case and use local variables? They aren't really relevant for the first test case anyways.

That would save some lines as you could directly define the ``results`` arrays in the objects instead of overwriting them later in the second test case. Actually, we can probably just leave out most of the data here. The code under test will only access ``results`` so we can try to keep it as minimal and concise as possible.

Example, this could be in the 2nd test case, or something along these lines at least:

```js
const submissionWithLock = { results: [new Result()] } as Submission;

const resultWithCompletionDate = new Result();
resultWithCompletionDate.completionDate = dayjs();
const unlockedSubmissionWithResult = { results: [resultWithCompletionDate] } as Submission;

...

const submissions = [submissionWithLock, unlockedSubmissionWithResult, ...];
```

For ``submissions`` in the first test case, you could just use an empty array. :D